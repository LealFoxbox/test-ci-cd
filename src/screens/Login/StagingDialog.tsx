import React from 'react';
import { Button, Dialog, Paragraph, Portal, useTheme } from 'react-native-paper';

import { styled } from 'src/paperTheme';

const NoButton = styled(Button)`
  margin-right: 10px;
`;

interface StagingDialog {
  onConfirm: () => void;
  hideDialog: () => void;
  visible: boolean;
}

const StagingDialog: React.FC<StagingDialog> = ({ onConfirm, visible, hideDialog }) => {
  const theme = useTheme();

  const handleConfirm = () => {
    hideDialog();
    onConfirm();
  };

  return (
    <Portal>
      <Dialog visible={visible} onDismiss={hideDialog}>
        <Dialog.Title>Alert</Dialog.Title>
        <Dialog.Content>
          <Paragraph>Are you sure you want to change environments?</Paragraph>
        </Dialog.Content>
        <Dialog.Actions>
          <NoButton color={theme.colors.placeholder} onPress={hideDialog}>
            Cancel
          </NoButton>
          <Button color={theme.colors.accent} onPress={handleConfirm}>
            Yes
          </Button>
        </Dialog.Actions>
      </Dialog>
    </Portal>
  );
};
export default StagingDialog;
