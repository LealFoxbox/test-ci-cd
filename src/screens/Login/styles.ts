import { styled } from 'src/paperTheme';

export const FormContainer = styled.View`
  padding: 50px;
  padding-top: 100px;
`;

export const EasterEgg = styled.View`
  width: 120px;
  height: 50px;
  position: absolute;
  left: 25px;
  top: 0;
  z-index: -99999;
`;
