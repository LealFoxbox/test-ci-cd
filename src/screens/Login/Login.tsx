import React, { useLayoutEffect, useState } from 'react';
import { TextInput as NativeTextInput, TouchableWithoutFeedback } from 'react-native';
import { Button, TextInput } from 'react-native-paper';
import { useMutation } from 'react-query';
import { AxiosResponse } from 'axios';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { useNavigation } from '@react-navigation/native';

import ErrorMessage from 'src/components/ErrorMessage';
import { ScrollView } from 'src/components/KeyboardAware';
import ConnectionBanner from 'src/components/ConnectionBanner';
import { UserResponse, authenticate } from 'src/services/api/user';
import config from 'src/config';
import { useNetworkStatus } from 'src/utils/useNetworkStatus';
import { PersistentUserStore } from 'src/pullstate/persistentStore';
import { ApiError } from 'src/services/api/utils';
import { clearInspectionsDataAction, loginAction, setStagingAction } from 'src/pullstate/actions';
import { styled } from 'src/paperTheme';

import StagingDialog from './StagingDialog';

export const FormContainer = styled.View`
  padding: 50px;
  padding-top: 100px;
`;

export const EasterEgg = styled.View`
  width: 100px;
  height: 50px;
`;

interface Form {
  companyId: string;
  username: string;
  password: string;
}

const SignInSchema = Yup.object().shape({
  companyId: Yup.string().required(`Please enter your Account Name.`),
  username: Yup.string().required(`Please provide your Email/Username.`),
  password: Yup.string().required(`Please enter your password.`),
});

const authenticateError = 'Your username or password appears to be incorrect for this account';

const LoginScreen: React.FC<{}> = () => {
  const isStaging = PersistentUserStore.useState((s) => s.isStaging);
  const [authError, setAuthError] = useState('');
  const connected = useNetworkStatus();
  const [visible, setVisible] = React.useState(false);
  const navigation = useNavigation();

  const [mutateSubmit, { isLoading }] = useMutation<AxiosResponse<UserResponse>, ApiError, Form>(authenticate, {
    onMutate: () => {
      setAuthError('');
    },
    onSuccess: async (userData) => {
      if (userData.status === 200 && userData.data.user) {
        setAuthError('');
        setStagingAction(isStaging);
        await clearInspectionsDataAction();
        loginAction(userData.data.user);
      } else {
        setAuthError(authenticateError);
      }
    },
    onError: (err) => {
      if (err?.message !== 'Network Error') {
        setAuthError(err?.response?.data.message || authenticateError);
      } else if (connected) {
        setAuthError(authenticateError);
      }
    },
  });

  useLayoutEffect(() => {
    navigation.setOptions({
      headerRight: (
        <TouchableWithoutFeedback delayLongPress={4000} accessibilityRole="none" onLongPress={() => setVisible(true)}>
          <EasterEgg />
        </TouchableWithoutFeedback>
      ),
    });
  }, [navigation]);

  const handleEasterEgg = () => {
    if (!isLoading) {
      setAuthError('');

      if (isStaging) {
        PersistentUserStore.update((s) => {
          s.isStaging = false;
        });
      } else {
        PersistentUserStore.update((s) => {
          s.isStaging = true;
        });
      }
    }
  };

  const clearError = <T extends string | React.ChangeEvent<any>>(onChange: (e: T) => void) => {
    return (e: T) => {
      setAuthError('');
      onChange(e);
    };
  };

  return (
    <ScrollView>
      <ConnectionBanner connected={connected} />
      <Formik
        initialValues={{
          companyId: config.isDev ? 'foxbox' : '',
          username: config.isDev ? 'diego' : '',
          password: config.isDev ? 'foxbox2020' : '',
        }}
        validationSchema={SignInSchema}
        onSubmit={(form) => mutateSubmit(form)}
      >
        {({ values, handleChange, errors, setFieldTouched, touched, isValid, handleSubmit }) => (
          <FormContainer>
            <TextInput
              value={values.companyId}
              onChangeText={clearError(handleChange('companyId'))}
              onBlur={() => setFieldTouched('companyId')}
              style={{ marginBottom: 10, textAlign: 'right' }}
              keyboardType="default"
              autoCapitalize="none"
              dense
              right={<TextInput.Affix text={`.${config.BACKEND_BASE_URL}`} />}
              /* @ts-ignore */
              render={(props) => <NativeTextInput {...props} textAlign="right" />}
              label="Account Name"
              disabled={isLoading}
              error={touched.companyId && !!errors.companyId}
            />
            {touched.companyId && errors.companyId && <ErrorMessage>{errors.companyId}</ErrorMessage>}
            <TextInput
              value={values.username}
              onChangeText={clearError(handleChange('username'))}
              onBlur={() => setFieldTouched('username')}
              style={{ marginBottom: 10 }}
              keyboardType="default"
              autoCapitalize="none"
              dense
              label="Email or Username"
              disabled={isLoading}
              error={touched.username && !!errors.username}
            />
            {touched.username && errors.username && <ErrorMessage>{errors.username}</ErrorMessage>}
            <TextInput
              value={values.password}
              onChangeText={clearError(handleChange('password'))}
              style={{ marginBottom: 10 }}
              keyboardType="default"
              autoCapitalize="none"
              dense
              autoCorrect={false}
              label="Password"
              onBlur={() => setFieldTouched('password')}
              secureTextEntry
              disabled={isLoading}
              onSubmitEditing={() => {
                if (isValid && !authError) {
                  handleSubmit();
                }
              }}
              error={(touched.password && !!errors.password) || !!authError}
            />
            {touched.password && (errors.password || !!authError) && (
              <ErrorMessage>{errors.password || authError}</ErrorMessage>
            )}
            <Button
              disabled={!isValid || !!authError || isLoading}
              onPress={handleSubmit}
              mode="contained"
              loading={isLoading}
              dark
              style={{ width: 120, alignSelf: 'flex-end', marginTop: 10 }}
            >
              Sign in
            </Button>
          </FormContainer>
        )}
      </Formik>
      <StagingDialog visible={visible} hideDialog={() => setVisible(false)} onConfirm={handleEasterEgg} />
    </ScrollView>
  );
};

export default LoginScreen;
