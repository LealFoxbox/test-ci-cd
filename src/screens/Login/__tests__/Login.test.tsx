import React from 'react';

import { act, render } from 'helpers/testUtils';

import Login from '../Login';

// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
const { useRoute } = jest.requireMock('@react-navigation/native');

describe('LoginScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
  });

  it('should render', async () => {
    useRoute.mockReturnValue({
      params: { updateRenderRight: () => undefined },
    });

    const result = render(<Login />);

    // eslint-disable-next-line @typescript-eslint/no-misused-promises
    await act(() => Promise.resolve());

    expect(result).toMatchSnapshot();
  });
});
