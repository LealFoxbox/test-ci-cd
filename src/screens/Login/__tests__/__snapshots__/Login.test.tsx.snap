// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render 1`] = `
Array [
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            pointerEvents="auto"
            removeClippedSubviews={false}
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                null,
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                accessibilityElementsHidden={false}
                importantForAccessibility="auto"
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                  ]
                }
              >
                <RCTScrollView
                  automaticallyAdjustContentInsets={false}
                  contentInset={
                    Object {
                      "bottom": 0,
                    }
                  }
                  enableAutomaticScroll={true}
                  enableOnAndroid={true}
                  enableResetScrollToCoords={true}
                  extraHeight={75}
                  extraScrollHeight={30}
                  getScrollResponder={[Function]}
                  handleOnScroll={[Function]}
                  keyboardDismissMode="interactive"
                  keyboardOpeningTime={250}
                  keyboardShouldPersistTaps="handled"
                  keyboardSpace={0}
                  onScroll={[Function]}
                  resetKeyboardSpace={[Function]}
                  scrollEventThrottle={1}
                  scrollForExtraHeightOnAndroid={[Function]}
                  scrollIntoView={[Function]}
                  scrollToEnd={[Function]}
                  scrollToFocusedInput={[Function]}
                  scrollToPosition={[Function]}
                  showsVerticalScrollIndicator={true}
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                  update={[Function]}
                  viewIsInsideTabBar={false}
                >
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "paddingBottom": 50,
                            "paddingLeft": 50,
                            "paddingRight": 50,
                            "paddingTop": 100,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "rgb(240, 240, 240)",
                              "borderTopLeftRadius": 4,
                              "borderTopRightRadius": 4,
                            },
                            Object {
                              "marginBottom": 10,
                              "textAlign": "right",
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "rgba(0, 0, 0, 0.26)",
                              "bottom": 0,
                              "height": 2,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "transform": Array [
                                Object {
                                  "scaleY": 0.5,
                                },
                              ],
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "minHeight": 52,
                              "paddingBottom": 0,
                              "paddingTop": 0,
                            }
                          }
                        >
                          <View
                            pointerEvents="none"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "transform": Array [
                                  Object {
                                    "translateX": 3,
                                  },
                                ],
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              onLayout={[Function]}
                              style={
                                Object {
                                  "color": "#FF951F",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "left": 0,
                                  "opacity": 0,
                                  "paddingLeft": 12,
                                  "paddingRight": 44,
                                  "position": "absolute",
                                  "textAlign": "left",
                                  "top": 28,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                    Object {
                                      "translateY": -10,
                                    },
                                    Object {
                                      "scale": 0.75,
                                    },
                                  ],
                                  "writingDirection": "ltr",
                                }
                              }
                            >
                              Account Name
                            </Text>
                            <Text
                              numberOfLines={1}
                              style={
                                Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "left": 0,
                                  "opacity": 0,
                                  "paddingLeft": 12,
                                  "paddingRight": 44,
                                  "position": "absolute",
                                  "textAlign": "left",
                                  "top": 28,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                    Object {
                                      "translateY": -10,
                                    },
                                    Object {
                                      "scale": 0.75,
                                    },
                                  ],
                                  "writingDirection": "ltr",
                                }
                              }
                            >
                              Account Name
                            </Text>
                          </View>
                          <TextInput
                            allowFontScaling={true}
                            autoCapitalize="none"
                            editable={true}
                            keyboardType="default"
                            multiline={false}
                            onBlur={[Function]}
                            onChangeText={[Function]}
                            onFocus={[Function]}
                            placeholder=""
                            placeholderTextColor="rgba(0, 0, 0, 0.54)"
                            rejectResponderTermination={true}
                            selectionColor="#FF951F"
                            style={
                              Array [
                                Object {
                                  "flexGrow": 1,
                                  "margin": 0,
                                  "textAlign": "left",
                                  "zIndex": 1,
                                },
                                Object {
                                  "paddingLeft": 12,
                                  "paddingRight": 44,
                                },
                                Object {
                                  "height": 52,
                                },
                                Object {
                                  "paddingBottom": 2,
                                  "paddingTop": 22,
                                },
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "textAlignVertical": "center",
                                },
                                Object {
                                  "marginRight": 0,
                                  "paddingRight": 36,
                                },
                              ]
                            }
                            textAlign="right"
                            underlineColorAndroid="transparent"
                            value="mobiletest"
                          />
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                              "opacity": 1,
                              "position": "absolute",
                              "right": 12,
                              "top": null,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                },
                                Object {
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                },
                              ]
                            }
                          >
                            .orangeqc-staging.com
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "rgb(240, 240, 240)",
                              "borderTopLeftRadius": 4,
                              "borderTopRightRadius": 4,
                            },
                            Object {
                              "marginBottom": 10,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "rgba(0, 0, 0, 0.26)",
                              "bottom": 0,
                              "height": 2,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "transform": Array [
                                Object {
                                  "scaleY": 0.5,
                                },
                              ],
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "minHeight": 52,
                              "paddingBottom": 0,
                              "paddingTop": 0,
                            }
                          }
                        >
                          <View
                            pointerEvents="none"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "transform": Array [
                                  Object {
                                    "translateX": 3,
                                  },
                                ],
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              onLayout={[Function]}
                              style={
                                Object {
                                  "color": "#FF951F",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "left": 0,
                                  "opacity": 0,
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                  "position": "absolute",
                                  "textAlign": "left",
                                  "top": 28,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                    Object {
                                      "translateY": -10,
                                    },
                                    Object {
                                      "scale": 0.75,
                                    },
                                  ],
                                  "writingDirection": "ltr",
                                }
                              }
                            >
                              Email or Username
                            </Text>
                            <Text
                              numberOfLines={1}
                              style={
                                Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "left": 0,
                                  "opacity": 0,
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                  "position": "absolute",
                                  "textAlign": "left",
                                  "top": 28,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                    Object {
                                      "translateY": -10,
                                    },
                                    Object {
                                      "scale": 0.75,
                                    },
                                  ],
                                  "writingDirection": "ltr",
                                }
                              }
                            >
                              Email or Username
                            </Text>
                          </View>
                          <TextInput
                            allowFontScaling={true}
                            autoCapitalize="none"
                            editable={true}
                            keyboardType="default"
                            multiline={false}
                            onBlur={[Function]}
                            onChangeText={[Function]}
                            onFocus={[Function]}
                            placeholder=""
                            placeholderTextColor="rgba(0, 0, 0, 0.54)"
                            rejectResponderTermination={true}
                            selectionColor="#FF951F"
                            style={
                              Array [
                                Object {
                                  "flexGrow": 1,
                                  "margin": 0,
                                  "textAlign": "left",
                                  "zIndex": 1,
                                },
                                Object {
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                },
                                Object {
                                  "height": 52,
                                },
                                Object {
                                  "paddingBottom": 2,
                                  "paddingTop": 22,
                                },
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "textAlignVertical": "center",
                                },
                                Array [
                                  Object {},
                                ],
                              ]
                            }
                            underlineColorAndroid="transparent"
                            value="diego"
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "rgb(240, 240, 240)",
                              "borderTopLeftRadius": 4,
                              "borderTopRightRadius": 4,
                            },
                            Object {
                              "marginBottom": 10,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "rgba(0, 0, 0, 0.26)",
                              "bottom": 0,
                              "height": 2,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "transform": Array [
                                Object {
                                  "scaleY": 0.5,
                                },
                              ],
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "minHeight": 52,
                              "paddingBottom": 0,
                              "paddingTop": 0,
                            }
                          }
                        >
                          <View
                            pointerEvents="none"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "transform": Array [
                                  Object {
                                    "translateX": 3,
                                  },
                                ],
                              }
                            }
                          >
                            <Text
                              numberOfLines={1}
                              onLayout={[Function]}
                              style={
                                Object {
                                  "color": "#FF951F",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "left": 0,
                                  "opacity": 0,
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                  "position": "absolute",
                                  "textAlign": "left",
                                  "top": 28,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                    Object {
                                      "translateY": -10,
                                    },
                                    Object {
                                      "scale": 0.75,
                                    },
                                  ],
                                  "writingDirection": "ltr",
                                }
                              }
                            >
                              Password
                            </Text>
                            <Text
                              numberOfLines={1}
                              style={
                                Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "left": 0,
                                  "opacity": 0,
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                  "position": "absolute",
                                  "textAlign": "left",
                                  "top": 28,
                                  "transform": Array [
                                    Object {
                                      "translateX": 0,
                                    },
                                    Object {
                                      "translateY": -10,
                                    },
                                    Object {
                                      "scale": 0.75,
                                    },
                                  ],
                                  "writingDirection": "ltr",
                                }
                              }
                            >
                              Password
                            </Text>
                          </View>
                          <TextInput
                            allowFontScaling={true}
                            autoCapitalize="none"
                            autoCorrect={false}
                            editable={true}
                            keyboardType="default"
                            multiline={false}
                            onBlur={[Function]}
                            onChangeText={[Function]}
                            onFocus={[Function]}
                            onSubmitEditing={[Function]}
                            placeholder=""
                            placeholderTextColor="rgba(0, 0, 0, 0.54)"
                            rejectResponderTermination={true}
                            secureTextEntry={true}
                            selectionColor="#FF951F"
                            style={
                              Array [
                                Object {
                                  "flexGrow": 1,
                                  "margin": 0,
                                  "textAlign": "left",
                                  "zIndex": 1,
                                },
                                Object {
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                },
                                Object {
                                  "height": 52,
                                },
                                Object {
                                  "paddingBottom": 2,
                                  "paddingTop": 22,
                                },
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                  "fontWeight": undefined,
                                  "textAlignVertical": "center",
                                },
                                Array [
                                  Object {},
                                ],
                              ]
                            }
                            underlineColorAndroid="transparent"
                            value="foxbox2020"
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignSelf": "flex-end",
                            "backgroundColor": "#FF951F",
                            "borderColor": "transparent",
                            "borderRadius": 4,
                            "borderStyle": "solid",
                            "borderWidth": 0,
                            "elevation": 2,
                            "marginTop": 10,
                            "minWidth": 64,
                            "shadowColor": "#000000",
                            "shadowOffset": Object {
                              "height": 0.75,
                              "width": 0,
                            },
                            "shadowOpacity": 0.24,
                            "shadowRadius": 1.5,
                            "width": 120,
                          }
                        }
                      >
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "disabled": false,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "borderRadius": 4,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              numberOfLines={1}
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontWeight": "400",
                                    "textAlign": "left",
                                  },
                                  Array [
                                    Object {
                                      "letterSpacing": 1,
                                      "marginHorizontal": 16,
                                      "marginVertical": 9,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                    Object {
                                      "textTransform": "uppercase",
                                    },
                                    Object {
                                      "color": "#ffffff",
                                      "fontFamily": "System",
                                      "fontWeight": "500",
                                    },
                                    Object {
                                      "fontFamily": "System",
                                      "fontWeight": "500",
                                    },
                                    undefined,
                                  ],
                                ]
                              }
                            >
                              Sign in
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </View>
          </View>
        </View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderTopColor": "rgb(216, 216, 216)",
              "borderTopWidth": 0.5,
              "bottom": 0,
              "elevation": 8,
              "height": 49,
              "left": 0,
              "paddingBottom": 0,
              "paddingHorizontal": 0,
              "position": null,
              "right": 0,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityLabel="ComponentToTest, tab, 1 of 1"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "selected": true,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "flex-end",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "rgb(0, 122, 255)",
                    },
                    Object {
                      "fontSize": 10,
                    },
                    undefined,
                  ]
                }
              >
                ComponentToTest
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaProvider>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />,
]
`;
